lib/action_view/component/base.rb

```diff
@@ -172,10 +172,12 @@ def identifier
         def templates
           @templates ||=
             (Dir["#{source_location.sub(/#{File.extname(source_location)}$/, '')}.*{#{ActionView::Template.template_handler_extensions.join(',')}}"] - [source_location]).each_with_object([]) do |path, memo|
+              pieces = File.basename(path).split(".")
+
               memo << {
                 path: path,
-                variant: path.split(".").second.split("+")[1]&.to_sym,
-                handler: path.split(".").last
+                variant: pieces.second.split("+")[1]&.to_sym,
```

Comment:
What happens if the path contains several folders with dots in them?

Reply:
💯 Great idea. I get a few tests in there for this edge case.

Reply:
One last question, what should happen if the file basename starts with a dot? I don't think we will ever encounter that scenario but better safe than sorry.